bindings/rust/s2n-tls-tokio/src/lib.rs

```diff
@@ -0,0 +1,135 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use errno::{set_errno, Errno};
+use s2n_tls::raw::{config::Config, connection::Connection, error::Error, ffi::s2n_mode};
+use std::{
+    future::Future,
+    os::raw::{c_int, c_void},
+    pin::Pin,
+    task::{Context, Poll},
+};
+use tokio::io::{AsyncRead, AsyncWrite, ReadBuf};
+
+const IO_ERROR_RETURN: c_int = -1;
+
+pub struct TlsAcceptor {
+    config: Config,
+}
+
+impl TlsAcceptor {
+    pub fn new(config: Config) -> Self {
+        TlsAcceptor { config }
+    }
+
+    pub async fn accept<S>(&self, stream: S) -> Result<TlsStream<S>, Error>
+    where
+        S: AsyncRead + AsyncWrite + Unpin,
+    {
+        TlsStream::open(self.config.clone(), s2n_mode::SERVER, stream).await
+    }
+}
+
+pub struct TlsConnector {
+    config: Config,
+}
+
+impl TlsConnector {
+    pub fn new(config: Config) -> Self {
+        TlsConnector { config }
+    }
+
+    pub async fn connect<S>(&self, _domain: &str, stream: S) -> Result<TlsStream<S>, Error>
```

Comment:
Because setting the server name is a connection method, not a config method, and the current interface doesn't support connection-level setup before the handshake. 

I'm considering adding functionality for calling connection methods to the config structure, or we could go the "with" route that rustls did: https://github.com/tokio-rs/tls/blob/master/tokio-rustls/src/lib.rs#L109-L133